dashboard/src/components/ProvisionButton/index.tsx

```diff
@@ -173,62 +116,40 @@ class ProvisionButton extends React.Component<IProvisionButtonProps, IProvisionB
     });
   };
 
-  public handleProvision = async (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    const { namespace, provision, push } = this.props;
+  public handleProvision = async ({ formData }: ISubmitEvent<any>) => {
+    const { namespace, provision, push, selectedClass, selectedPlan } = this.props;
     this.setState({ isProvisioning: true });
-    const { releaseName, selectedClass, selectedPlan, parameters } = this.state;
 
-    const parametersObject = JSON.parse(parameters);
+    const { name, parameters, ...rest } = formData;
     if (selectedClass && selectedPlan) {
       const provisioned = await provision(
-        releaseName,
+        name,
         namespace,
         selectedClass.spec.externalName,
         selectedPlan.spec.externalName,
-        parametersObject,
+        parameters || rest,
```

Comment:
out of curiosity, might be cases in which that the hash `parameters` and other properties are set (...rest)?

Reply:
`parameters` is used when the broker doesn't provide a schema (instanceCreateParameterSchema is null). It's possible that the broker schema defines a key called `parameters` which would conflict with this (probably what you're referring to). I don't think this is very likely, but we could rename this to `kubeappsParameters` or something a bit more unique to be more sure. WDYT?

Reply:
yes, it would be better to have `kubeappsParameters` just to be sure.

Reply:
@andresmgot updated - PTAL!